internal/testutils/kginkgo.go

```diff
@@ -0,0 +1,230 @@
+package testutils
+
+import (
+	"context"
+	"time"
+
+	ngrokv1alpha1 "github.com/ngrok/ngrok-operator/api/ngrok/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	DefaultTimeout  = 20 * time.Second
+	DefaultInterval = 500 * time.Millisecond
+)
+
+// KGinkgo is a helper for Ginkgo tests that interact with Kubernetes resources.
+// It provides methods to assert conditions on Kubernetes objects using Gomega matchers, especially in conjunction with Eventually.
+type KGinkgo struct {
+	client client.Client
```

Comment:
want to collapse these functions i made into this struct or do that later?
https://github.com/ngrok/ngrok-operator/blob/main/internal/testutils/ginkgo.go

Reply:
Yep. I'll fix it up as part of my rebase onto the latest main